DEVELOPER.md

```diff
@@ -49,286 +13,326 @@ To test and run services, you need to get access to the development Kubernetes c
 Please see [Kubernetes](https://kubernetes.io) for setting up a development cluster.
 
-Please read [doc](https://www.fluvio.io) for a technical arch and operation guide. 
+Please read [doc](https://www.fluvio.io) for a technical arch and operation guide.
 
 ---
 
-## Setting up Development Environment
+## Requirements
 
-The development environment requires the Rust toolchain and a few other build dependencies to build changes to the source code.
+You will need the Rust Toolchain to build Fluvio, follow [Rust Installation](https://www.rust-lang.org/tools/install) guide to get Rust
+Toolchain running on your system.
 
-To deploy and test the source code, Kubernetes is required because Fluvio stores its metadata in Kubernetes. Fluvio binaries can be tested locally, running outside of any k8s cluster using the `--local` parameter or they can be built into a docker image and deployed into the Kubernetes cluster.
+Other required Software that might be installed in your system includes:
 
-### Rust toolchain
+- [make](https://www.gnu.org/software/make/)
+- [git](https://git-scm.com/)
 
+## Building
 
-Please follow [setup](https://www.rust-lang.org/tools/install) instructions to install Rust and Cargo.
+It is recommended to use native binaries for development and testing but not for production.
 
-### Build time dependencies
+The following section will guide you through building different Fluvio
+components.
 
-Most of these are required for building a docker image: it is also possible to not install these and just `cargo build` the binaries you need and test them with a local setup.
+### Fluvio CLI
 
-* make
-* zig
-* lld (v16)
-* git
+Fluvio CLI is the main entry point to interact with Fluvio cluster.
+It is required to install, manage and access the Fluvio cluster.
 
-### Kubernetes dependencies
+Use the following command to build the CLI binary with all features:
 
-Kubernetes is required for Fluvio to store its metadata.
+```bash
+$ make build-cli
+```
 
-Please set up one of the following recommended kubernetes distros:
-
-* [Rancher desktop](https://rancherdesktop.io)
-* [k3d](https://k3d.io)
-* [kind](https://kind.sigs.k8s.io)
+### Fluvio Cluster
 
+Entire Fluvio cluster (including SC and SPU) is contained a single binary,
+use the following command to build the cluster binary:
 
-#### Helm
+```bash
+$ make build-cluster
+```
 
-Helm is used for installing Fluvio on Kubernetes.
+## Development
 
-Please follow [helm setup](https://helm.sh/docs/intro/quickstart/) to install the helm.
+### Running Fluvio Cluster Locally
 
+Use following commands to start Fluvio cluster using native binaries.
 
-### Linker Pre-requisites
+> [!NOTE]
+> Following sections will use `flvd` alias to refer to the locally
+> builded Fluvio CLI.
+> Learn more on how to set it up [here](#aliasing-fluvio-development-binaries).
 
-Zig and LLVM is required. 
+```bash
+$ flvd cluster start --develop
```

Comment:
do we still need develop flag?

Reply:
I dont think so, unless you want to use the local build image.

Reply:
Just removed `--develop` instances.